src/vivarium/framework/results/interface.py

```diff
@@ -438,7 +436,7 @@ def register_concatenating_observation(
 
     @staticmethod
     def _check_for_required_callables(
-        observation_name: str, required_callables: dict[str, Callable]
+        observation_name: str, required_callables: dict[str, Callable[..., pd.DataFrame]]
```

Comment:
This makes me think we _should_ properly type hint the placeholder callable; we can definitely do better than `...`, no?

Reply:
I agree, we should type-hint the placeholder, since we do know something about the arguments the provided callable should have.